applications/GeoMechanicsApplication/custom_elements/geo_steady_state_Pw_piping_element.h

```diff
@@ -140,6 +143,80 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoSteadyStatePwPipingElement : publ
                std::abs(head_gradient);
     }
 
+    void CalculateOnIntegrationPoints(const Variable<bool>& rVariable,
+                                      std::vector<bool>&    rValues,
+                                      const ProcessInfo&    rCurrentProcessInfo) override
+    {
+        if (rVariable == PIPE_ACTIVE) {
+            const auto number_of_integration_points =
+                this->GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
+            rValues.resize(number_of_integration_points);
+            std::fill_n(rValues.begin(), number_of_integration_points, this->GetValue(rVariable));
+        }
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<double>& rVariable,
+                                      std::vector<double>&    rValues,
+                                      const ProcessInfo&      rCurrentProcessInfo) override
+    {
+        if (rVariable == PIPE_HEIGHT) {
+            const auto number_of_integration_points =
+                this->GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
+            rValues.resize(number_of_integration_points);
+            std::fill_n(rValues.begin(), number_of_integration_points, this->GetValue(rVariable));
+        }
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<array_1d<double, 3>>& rVariable,
```

Comment:
I have some suggestions on how we might simplify the body of this function. Perhaps we can discuss that when you are available?

Reply:
certainly.